tests/core/test_reference.py

```diff
@@ -161,7 +161,7 @@ class InnerSchema(BaseModel):
     class MediumSchema(BaseModel):
         s: str
         nested: InnerSchema
-        nested_opt: Optional[InnerSchema]
```

Comment:
Note: I had to change this in the test suite. For some reason, leaving it as `Optional[InnerSchema]` caused `ref.nested.nested_opt.m.uuid` not to be raised. Must be a change in how Pydantic 2 handles optional-annotated fields.